rootstrap/node-ts-api-base
# Feature/authentication

This PR contains:

- [X] Routes & controllers for authentication (sign in & sign up).
- [X] Complete testing for user's controller.
- [X] Testing for sign in & sign up.
- [X] Middleware for restricted routes based on JWT.

Note: [This is using JWT](https://jwt.io/) on the midtime, this c tan be changed in the future to support sign out feature.

**Testing steps (NOTES):**

- Take a look at `.example.env.test` & `.example.env`, there are some new changes to take into consideration.
- Make sure you have a database for testing with the same name as your `TYPEORM_DATABASE` env variable.
- Since `TYPEORM_SYNCHRONIZE` is settled as true it should run all needed changes to the DB when testing starts (`yarn test`)

![Screen Shot 2020-09-18 at 09 19 26](https://user-images.githubusercontent.com/8084303/93596648-1324b100-f990-11ea-87c4-722489744cc5.png)
